extension/js/common/platform/store/contact-store.ts

```diff
@@ -272,38 +354,31 @@ export class ContactStore extends AbstractStore {
         return resultsWithPgp.concat(resultsWithoutPgp);
       }
     }
-    const raw: unknown[] = await new Promise((resolve, reject) => {
-      const contacts = db.transaction('contacts', 'readonly').objectStore('contacts');
+    const emails = db.transaction(['emails'], 'readonly').objectStore('emails');
+    const raw: Email[] = await new Promise((resolve, reject) => {
       let search: IDBRequest;
       if (typeof query.has_pgp === 'undefined') { // any query.has_pgp value
-        search = contacts.openCursor(); // no substring, already covered in `typeof query.has_pgp === 'undefined' && query.substring` above
+        search = emails.openCursor(); // no substring, already covered in `typeof query.has_pgp === 'undefined' && query.substring` above
       } else { // specific query.has_pgp value
-        if (query.substring) {
-          search = contacts.index('search').openCursor(IDBKeyRange.only(ContactStore.dbIndex(query.has_pgp, query.substring)));
-        } else {
-          search = contacts.index('index_has_pgp').openCursor(IDBKeyRange.only(Number(query.has_pgp)));
-        }
+        const indexRange = ContactStore.dbIndexRange(query.has_pgp, query.substring ?? '');
+        const range = IDBKeyRange.bound(indexRange.lowerBound, indexRange.upperBound, false, true);
```

Comment:
Please add comments explaining exactly how this range works, how we (ab)use it and why. I think you explained it well in an earlier comment on the issue or PR, can just copy to become code comment.

Reply:
Done